src/main/scala/ch/epfl/bluebrain/nexus/kg/resources/Files.scala

```diff
@@ -67,6 +68,69 @@ class Files[F[_]: Effect: Timer](repo: Repo[F])(implicit config: AppConfig) {
       saveStorage: Save[F, In]): RejOrResource[F] =
     repo.updateFile(id, storage, rev, fileDesc, source)
 
+  /**
+    * Creates a link to an existing file.
+    *
+    * @param projectRef reference for the project in which the resource is going to be created.\
+    * @param base       base used to generate new ids
+    * @param storage    the storage where the file is going to be saved
+    * @param fileDesc   the file description metadata
+    * @param location   the file location URI
+    * @return either a rejection or the new resource representation in the F context
+    */
+  def createLink(projectRef: ProjectRef, base: AbsoluteIri, storage: Storage, fileDesc: FileDescription, location: Uri)(
+      implicit subject: Subject,
+      linkStorage: Link[F]): RejOrResource[F] =
+    createLink(Id(projectRef, generateId(base)), storage, fileDesc, location)
+
+  /**
+    * Creates a link to an existing file.
+    *
+    * @param id       the id of the resource
+    * @param storage  the storage where the file is going to be saved
+    * @param fileDesc the file description metadata
+    * @param location the file location URI
+    * @return either a rejection or the new resource representation in the F context
+    */
+  def createLink(id: ResId, storage: Storage, fileDesc: FileDescription, location: Uri)(
+      implicit subject: Subject,
+      linkStorage: Link[F]): RejOrResource[F] =
+    repo.createLink(id, storage, fileDesc, location)
+
+  /**
+    * Updates a link to an existing file.
+    *
+    * @param projectRef reference for the project in which the resource is going to be created.\
+    * @param base       base used to generate new ids
+    * @param storage    the storage where the file is going to be saved
+    * @param fileDesc   the file description metadata
+    * @param location   the file location URI
+    * @param rev        the last known resource revision
+    * @return either a rejection or the new resource representation in the F context
+    */
+  def updateLink(projectRef: ProjectRef,
```

Comment:
What is this used for? When updating a resource in nexus you always have the @id and you never autogenerate a new one.